### PR TITLE
feat(ui): add canonical next-action guidance

### PR DESCRIPTION
**Pull request by cryppadotta**
## Thinking Path

> - Paperclip is the open-source control plane people use to supervise agents and the work they own.
> - Operators currently have to infer an issue's next action from status, runs, interactions, blockers, and recovery state spread across several surfaces.
> - That ambiguity is most costly when work looks blocked even though it has a live continuation, or looks active even though it needs recovery or a human decision.
> - Paperclip needs one deterministic precedence rule that classifies the next action as active work, recovery, waiting, or a real blocker.
> - The same rule must drive issue detail, subtree diagnostics, and run views so those surfaces cannot disagree.
> - Successful-run handoff recovery also needs to stay on the normal agent lane without giving status-only recovery authority to create interactions or decompose plans.
> - This pull request adds the shared diagnostic contract, canonical resolver, three UI surfaces, bounded recovery behavior, and regression coverage.
> - The benefit is an inspectable next step for operators without hiding productive continuations or creating recovery loops.

## Linked Issues or Issue Description

Related public work: #10184 established normal-model successful-run handoff; #10628 improves live blocker coverage; #10671 maintains review-action paths. This pull request builds a consistent operator-facing next-action layer across those kinds of signals rather than replacing those efforts.

**Problem**

Issue status alone does not tell an operator who should act next. A blocked issue can still have productive work in flight, an `in_progress` issue can need recovery, and a descendant can be waiting for a human decision while its parent appears generically stalled. Different UI surfaces currently require operators to reconstruct that state independently.

**Expected behavior**

Paperclip should resolve exactly one prioritized next-action lane from company-scoped diagnostics: active work first, then recovery, then waiting, then a real blocker. Issue detail, subtree summaries, and run-touched issues should use the same resolver and explain the evidence behind the verdict. Missing-disposition recovery should use a bounded normal-model handoff and must not grant status-only recovery runs authority to create interactions or decompose accepted plans.

**Scope**

This change adds read-only diagnostic endpoints and UI guidance plus recovery safeguards. It does not add a schema migration, change issue status semantics, or introduce a new roadmap capability.

## What Changed

- Added shared company-scoped blocker and subtree diagnostic contracts plus issue API clients and query hooks.
- Added a deterministic four-lane next-action resolver with focused precedence, wording, and edge-case tests.
- Added token-compliant next-action cards to issue detail, subtree diagnostics, and run-touched issue rows.
- Routed ambiguous successful-run disposition follow-up through a fingerprinted normal-model handoff.
- Prevented status-only recovery runs from creating interactions or decomposing accepted plans directly.
- Documented the canonical next-action and recovery authority invariants.
- Added Storybook coverage and an optional screenshot helper for visual inspection.

## Verification

- `pnpm exec vitest run ui/src/lib/next-action.test.ts ui/src/components/IssueNextActionCard.test.tsx ui/src/components/IssueSubtreeNextActionView.test.tsx server/src/services/recovery/successful-run-handoff.test.ts server/src/__tests__/issue-agent-mutation-ownership-routes.test.ts server/src/__tests__/issue-blocker-diagnostics-routes.test.ts` — 135 tests passed.
- `pnpm check:token-gates` — passed.
- `pnpm -r typecheck` — passed for all workspace packages.
- `pnpm test:run` — 3,449 tests passed; the sole worktree-suppressed wake test passed on isolated rerun with `PAPERCLIP_IN_WORKTREE=false`.
- `pnpm build` — passed for all workspace packages.
- `git diff --check public-gh/master...HEAD` — passed.

## Risks

- Incorrect precedence could hide the signal an operator needs. Resolver tests cover active, recovery, waiting, blocker, descendant, and fallback combinations.
- Diagnostic reads add route work on issue detail and subtree surfaces. Queries remain company-scoped and are only enabled when their identifiers are present.
- Recovery handoff changes can affect wake behavior. Stable fingerprints and existing attempt bounds prevent repeated handoff loops.
- The UI introduces new operator wording. Storybook and render tests cover representative states, but final phrasing may benefit from maintainer feedback.

> This aligns with the completed “Enforced Outcomes” and “Self-healing runs & automatic recovery” roadmap items. It does not duplicate a planned core capability.

## Model Used

- OpenAI Codex using `gpt-5.6-sol`, with agentic reasoning, repository inspection, code execution, and GitHub CLI access. The runtime does not expose the context-window size.

## Checklist

- [x] I have included a thinking path that traces from project context to this change
- [x] I have specified the model used (with version and capability details)
- [x] I have checked ROADMAP.md and confirmed this PR does not duplicate planned core work
- [x] I have searched GitHub for duplicate or related PRs and linked them above
- [x] I have either (a) linked existing issues with `Fixes: #` / `Closes: #` / `Refs: #` OR (b) described the issue in-PR following the relevant issue template
- [x] I have not referenced internal/instance-local Paperclip issues or links (only public GitHub `#NNN` / `github.com/paperclipai/paperclip` URLs)
- [x] My branch name describes the change (e.g. `docs/...`, `fix/...`) and contains no internal Paperclip ticket id or instance-derived details
- [x] I have run tests locally and they pass
- [x] I have added or updated tests where applicable
- [x] I have updated relevant documentation to reflect my changes
- [x] I have considered and documented any risks above
- [x] All Paperclip CI gates are green
- [x] Greptile is 5/5 with no open P2s, recommendations, or follow-ups
- [x] I will address all Greptile and reviewer comments before requesting merge

### CODE REVIEW

File: ui/src/lib/next-action.ts [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+        laneLabel: "Recovery debt",
+        statement: "Recovery debt → assign worker",
+        target: blockerRef(gate),
+        gate: gateLabel(gate),
+        ready: false,
+        resolvedFrom: "cancelled_blocker_in_set",
+      };
+    }
+    if (gate.flags.includes("workspace_finalize_pending")) {
+      return {
```
[the remark this answers is not in the document]
Comment (cryppadotta) [reply ui/src/lib/next-action.ts:560]: Fixed in a9b8cf2213. Active-looking subtree statuses require queued/claimed wake evidence, and actionable selection is restricted to leaves. Stale active nodes now report missing continuation.

File: ui/src/lib/next-action.ts [hunk 2]
```diff
@@ -0,0 +1,575 @@
...
+  if (!diagnostics) return [];
+  return diagnostics.blockers.filter((blocker) =>
+    blocker.flags.some(
+      (flag) =>
+        flag === "done_but_blocking"
+        || flag === "workspace_finalize_pending"
+        || flag === "cancelled_blocker_in_set",
+    ),
+  );
+}
```
[the remark this answers is not in the document]
Comment (cryppadotta) [reply ui/src/lib/next-action.ts]: Fixed in a9b8cf2213. Only workspace_finalize_pending is a terminal gate now. Done/cancelled blocker edges are recovery debt with clear-or-replace guidance and distinct relation labels.

File: ui/src/components/IssueNextActionCard.tsx [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+      data-next-action-accent={summary.accent}
+      data-terminal-gate={summary.terminalGate ? "true" : undefined}
+      data-recovery-debt={summary.recoveryDebt ? "true" : undefined}
+      style={{ borderLeftColor: accent, borderLeftWidth: "4px" }}
+      className={`rounded-md border border-border bg-card text-card-foreground px-3 py-2.5 shadow-sm ${className ?? ""}`}
+    >
+      {/* 1. Lane chip */}
+      <div className="flex items-center gap-1.5">
+        {summary.live ? (
+          <span className="relative flex h-2 w-2" aria-hidden>
```
[the remark this answers is not in the document]
Comment (cryppadotta) [reply ui/src/components/IssueNextActionCard.tsx:133]: Fixed in a9b8cf2213. Border widths, indentation, marker surfaces, and marker shadow now resolve through tokens in ui/src/index.css; the token gate is clean.

File: ui/src/pages/AgentDetail.tsx [hunk 4]
```diff
@@ -3421,17 +3422,23 @@ function RunDetail({ run: initialRun, agentRouteId, adapterType, adapterConfig }
...
+                  <div className="flex items-center gap-2 min-w-0">
+                    <StatusBadge status={issue.status} />
+                    <span className="truncate">{issue.title}</span>
+                  </div>
+                  <span className="font-mono text-muted-foreground shrink-0 ml-2">{issue.identifier ?? issue.issueId.slice(0, 8)}</span>
+                </Link>
+                <RunNextActionVerdict
+                  issueId={issue.issueId}
+                  runIsActive={run.status === "running" || run.status === "queued"}
+                  className="mt-2"
```
[the remark this answers is not in the document]
Comment (cryppadotta) [reply ui/src/pages/AgentDetail.tsx:3440]: Fixed in a9b8cf2213. Active state is now true only when the live run's context scope names this touched task; focused tests cover current versus historical task IDs.
